skimage/filters/_window.py

```diff
@@ -0,0 +1,120 @@
+    all of the window types available to that function
+    (e.g., ``"hann"``, ``"boxcar"``). Note that certain window types require
+    parameters that have to be supplied with the window name as a tuple
+    (e.g., ``("tukey", 0.8)``). If only a float is supplied, it is interpreted
+    as the beta parameter of the Kaiser window.
+
+    See https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.windows.get_window.html
+    for more details.
+
+    Note that this function can generate very large arrays that consume
+    a large amount of available memory. For example, if ``size=512`` and
+    ``ndim=4`` are given as parameters to `skimage.filters.window`, it will
+    attempt to return an > 8.5GB array.
+
+    The approach taken here to create nD windows is to first calculate the
+    Euclidean distance from the center of the intended nD window to each
+    position in the array. That distance is used to sample, with
+    interpolation, from a 1D window returned from ``scipy.signal.get_window``.
+    The method of interpolation can be changed with the ``order`` keyword
+    argument passed to `skimage.transform.warp`.
+
+    Some coordinates in the output window will be outside of the original
+    signal; these will be filled in with zeros.
+
+    Window types:
+    - boxcar
+    - triang
+    - blackman
+    - hamming
+    - hann
+    - bartlett
+    - flattop
+    - parzen
+    - bohman
+    - blackmanharris
+    - nuttall
+    - barthann
+    - kaiser (needs beta)
+    - gaussian (needs standard deviation)
+    - general_gaussian (needs power, width)
+    - slepian (needs width)
+    - dpss (needs normalized half-bandwidth)
+    - chebwin (needs attenuation)
+    - exponential (needs decay scale)
+    - tukey (needs taper fraction)
+
+    Examples
+    --------
+    Return a Hann window with shape (512, 512):
+
+    >>> from skimage.filters import window
+    >>> w = window('hann', 512)
+
+    Return a Kaiser window with beta parameter of 16 and shape (256, 256, 256):
+
+    >>> w = window(16, 256, ndim=3)
+
+    Return a Tukey window with an alpha parameter of 0.8 and shape (100, 100):
+
+    >>> w = window(('tukey', 0.8), 100)
+
+    References
+    ----------
+    .. [1] Two-dimensional window design, Wikipedia,
+           https://en.wikipedia.org/wiki/Two_dimensional_window_design
+    """
+
+    ndim = safe_as_int(ndim)
+    size = safe_as_int(size)
+
+    if ndim <= 0:
+        raise ValueError("Number of dimensions must be greater than zero")
+
+    w = get_window(window_type, size, fftbins=False)
+    w = np.reshape(w, (-1,) + (1,) * (ndim-1))
+
+    # Create coords for warping following `ndimage.map_coordinates` convention.
+    L = [np.arange(size, dtype=np.double) for i in range(ndim)]
+    coords = np.stack((np.meshgrid(*L)))
+    center = (size / 2) - 0.5
+    coords[0, ...] = np.sqrt(((coords - center) ** 2).sum(axis=0)) + center
+    coords[1:, ...] = 0
```

Comment:
The following is more efficient and has lower memory use at cost of being more verbose. The key component is using the `sparse` argument to meshgrid which gives much smaller broadcastable coordinate arrays.


```Python
    center = (size / 2) - 0.5
    dist = 0
    for g in np.meshgrid(*L, sparse=True):
        g -= center
        dist = dist + g * g
    dist = np.sqrt(dist)
    coords = np.zeros((ndim,) + dist.shape, dtype=np.double)
    coords[0] = dist + center
```

It may also be sufficient to just use `np.float32` for the `coords` array.